### ISSUE TITLE
Add CREATE endpoint for tasks

### ISSUE THREAD

**Issue by anarute** [Igalia/phpreport#677]
Add POST endpoint for tasks.

Tasks have a few mandatory fields:

- date
- init
- end
- project 
- user

When creating a new task, we need to do a few checks:

- the end time needs to be after init.
- a task cannot overlap another existing task, that means a user shouldn't be allowed to create a tasks where the init or end times are between the init and end time of another task in the same day.
- user can create tasks in that date. This means the date is not blocked by the application admin. This is configured by the settings `dayLimitEnabled`, `numberOfDays` and `date`.
- project must be still active.

**Comment by dmtrek14** [comment]
We also need to ensure that if a task_type is submitted it exists in the task_type table
